### PR TITLE
feat(ui5-title): add custom overstyles when used with Link

### LINKED ISSUE #6785: Title - Title as a Link
### Bug Description

The title should be able to be set as a link while retaining most of the header style

### Affected Component

_No response_

### Expected Behaviour

image

### Isolated Example

https://sap.github.io/ui5-webcomponents/playground/components/Title/

### Steps to Reproduce

1.
2.
3.
...

### Log Output, Stack Trace or Screenshots

_No response_

### Priority

Medium

### UI5 Web Components Version

1.11.0

### Browser

Chrome

### Operating System

_No response_

### Additional Context

_No response_

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

### PR DESCRIPTION
**Pull request by gmkv**
There are cases when a Link will be used inside a Title.
Elements nested inside, mainly the Link, will inherit the Title's font styles.

Refactored Link to allow for overstyling. Updated text-underline when hovering a Subtle link to match the latest VD spec.

Fixes #6785

### CODE REVIEW

**Review by tsanislavgatev — APPROVED** [review]
+1 on link.

**Review by dimovpetar — CHANGES_REQUESTED** [review]
Also update the documentation of the slot sap.ui.webc.main.Title.prototype.default

**Review by dimovpetar — APPROVED** [review]

File: packages/main/test/pages/Title.html [hunk 1]
```diff
@@ -26,10 +27,32 @@
 		<ui5-title level="H2">Very long long long text of title title with level two</ui5-title>
 	</section>
 
-	<section class="title2auto">
+	<section class="wrapping">
 		<h2>Wrapping</h2>
-		<ui5-title level="H6" id="truncated-title" class="title3auto">Truncated title Lorem ipsum dolor, sit amet consectetur adipisicing elit. Numquam, ab.</ui5-title>
+		<ui5-title level="H6" id="truncated-title">Truncated title Lorem ipsum dolor, sit amet consectetur adipisicing elit. Numquam, ab.</ui5-title>
```
Comment (BorisDafov) [inline packages/main/test/pages/Title.html:32]: Truncated Title Lorem Ipsum Dolor, Sit Amet Consectetur Adipisicing Alit. Numquam, Ab.

File: packages/main/test/pages/Title.html [hunk 2]
```diff
@@ -26,10 +27,32 @@
...
 		<ui5-title level="H6" wrapping-type="Normal" id="wrapping-title">Wrapping title Lorem ipsum dolor, sit amet consectetur adipisicing elit. Numquam, ab.</ui5-title>
```
Comment (BorisDafov) [inline packages/main/test/pages/Title.html:33]: Wrapping Title Lorem Ipsum Dolor, Sit Amet Consectetur Adipisicing Elit. Numquam, Ab.

File: packages/main/test/pages/Title.html [hunk 3]
```diff
@@ -26,10 +27,32 @@
...
 	</section>
+
+	<section>
+		<h2>Usage with ui5-link</h2>
+		<ui5-title level="H3">
```
Comment (BorisDafov) [inline packages/main/test/pages/Title.html:38]: <ui5-title level="H3">
			<ui5-link design="Default">With Default Link (57)</ui5-link>
		</ui5-title>
		<ui5-title level="H3">
			<ui5-link design="Emphasized">With Emphasized Link (58)</ui5-link>
		</ui5-title>
		<ui5-title level="H3">
			<ui5-link design="Subtle">With Subtle Link (59)</ui5-link>
		</ui5-title>
		<div class="wrapping">
			<h2>Wrapping with Link</h2>
			<ui5-title level="H6">
				<ui5-link design="Default">Truncated with Link (60) Lorem Ipsum Dolor, Sit Amet Consectetur Adipisicing Elit. Numquam, Ab.</ui5-link>
			</ui5-title>
			<ui5-title level="H6" wrapping-type="Normal">
				<ui5-link design="Default">Wrapping with Link (61) Lorem Ipsum Dolor, Sit Amet Consectetur Adipisicing Elit. Numquam, Ab.</ui5-link>
			</ui5-title>

**Review by BorisDafov — CHANGES_REQUESTED** [review]

**Review by BorisDafov — COMMENTED** [review]
From UA perspective, I approve these changes...

**Review by dimovpetar — APPROVED** [review]

**Pull request merged**